### PR TITLE
Refactor and merge Cydex code to support adding other datasets

### PR DESCRIPTION
**Pull request by edwinzhng**
This refactor adds the Cydex code into Covidex to easily deploy with the ACL Anthology, and also makes it much easier to integrate new datasets using the following two steps:

1. Provide a Lucene index schema for the new dataset under `api/schema`
2. Create new components under the `client/src/components/dataset` folder for displaying the search result, and provide a `metadata.json` file

Documentation to come in follow up PR

### CODE REVIEW

File: api/app/services/searcher.py [hunk 1]
```diff
@@ -2,36 +2,36 @@
...
 
 
 class Searcher:
     def __init__(self):
-        self.searchers: List[pysearch.SimpleSearcher] = {}
-        self.searchers[SearchVertical.cord19] = self.build_searcher(
-            settings.cord19_index_path)
-        #self.searchers[SearchVertical.trialstreamer] = self.build_searcher(
-        #    settings.trialstreamer_index_path)
+        self.build_searcher(settings.index_path)
```
Comment (shaneding) [inline api/app/services/searcher.py:10]: if `self.searchers` is no longer a list, does it mean that covidex only supports searching 1 corpus atm?

File: client/src/components/pages/HomePage/HomePage.tsx [hunk 2]
```diff
@@ -5,85 +5,105 @@ import ErrorBoundary from 'react-error-boundary';
...
+  let includeArticle = true;
+  const fields = Object.keys(selectedFilters);
+  fields.forEach((field) => {
+    const val = article[field];
+    if (filterSchema[field].type === 'year_slider') {
+      includeArticle =
+        includeArticle &&
+        (!val ||
+          (val.length >= 7 &&
+            Number(val.substr(0, 7).replace('-', '')) >= selectedFilters[field][0] &&
```
Comment (shaneding) [inline client/src/components/pages/HomePage/HomePage.tsx:73]: Is this month and year based filtering?

Comment (edwinzhng) [reply client/src/components/pages/HomePage/HomePage.tsx:73]: Yeah only if month is available, we do this for covidex if the year range is small

File: api/app/services/searcher.py (hunk 1 above)
Comment (edwinzhng) [reply api/app/services/searcher.py:10]: Yeah, easy to change later but this is cleaner

Comment (edwinzhng) [reply api/app/services/searcher.py:10]: I don't think any deployment would need multiple corpora at the same time

**Comment by shaneding** [comment]
LGTM!

File: client/tsconfig.json [hunk 3]
```diff
@@ -21,5 +21,11 @@
   },
   "include": [
     "src"
+  ],
+  "plugins": [
+    {
+      "name": "typescript-styled-plugin",
```
Comment (edwinzhng) [inline client/tsconfig.json:27]: Ignoring this linter to avoid problems with CSS-in-JS

**Review by shaneding — APPROVED** [review]
lgtm

**Pull request merged**